java/src/main/java/com/konfigthis/splitit/client/ApiClient.java

```diff
@@ -165,18 +167,16 @@ public ApiClient(String basePath, String clientId, String clientSecret, Map<Stri
         if (basePath != null) {
             this.basePath = basePath;
         }
-
-        String tokenUrl = "https://id.production.splitit.com/connect/token";
-        if (!"".equals(tokenUrl) && !URI.create(tokenUrl).isAbsolute()) {
+        if (!"".equals(tokenPath) && !URI.create(tokenPath).isAbsolute()) {
             URI uri = URI.create(getBasePath());
-            tokenUrl = uri.getScheme() + ":" +
+            tokenPath = uri.getScheme() + ":" +
```

Comment:
is this right actually? I dont want to overwrite the instance variable do I?

Reply:
oh no its fine, after the scheme gets added once then it wont anymore

Reply:
when is this constructor even used?

Reply:
Shouldnt every constructor have to go through this one ultimately in order to set up oauth?

Reply:
actually, I'm a little confused. there's one other constructor that calls `authentications.put(...)`, but it doesn't reference the tokenUrl at all?